lib/ansible/module_utils/network/junos/config/vlans/vlans.py

```diff
@@ -0,0 +1,208 @@
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        root = build_root_xml_node('vlans')
+        state = self._module.params['state']
+        if state == 'overridden':
+            config_xmls = self._state_overridden(want, have)
+        elif state == 'deleted':
+            config_xmls = self._state_deleted(want, have)
+        elif state == 'merged':
+            config_xmls = self._state_merged(want, have)
+        elif state == 'replaced':
+            config_xmls = self._state_replaced(want, have)
+
+        for xml in config_xmls:
+            root.append(xml)
+
+        return tostring(root)
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the xml necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        intf_xml = []
+        intf_xml.extend(self._state_deleted(want, have))
+        intf_xml.extend(self._state_merged(want, have))
+        return intf_xml
+
+    def _state_overridden(self, want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the xml necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        intf_xml = []
+        intf_xml.extend(self._state_deleted(have, have))
+        intf_xml.extend(self._state_merged(want, have))
+        return intf_xml
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the xml necessary to merge the provided into
+                  the current configuration
+        """
+        intf_xml = []
+
+        for config in want:
+            vlan_name = str(config['name'])
+            vlan_id = str(config['vlan_id'])
+            vlan_description = config.get('description')
+            vlan_root = build_root_xml_node('vlan')
+            build_child_xml_node(vlan_root, 'name', vlan_name)
+            build_child_xml_node(vlan_root, 'vlan-id', vlan_id)
+            if vlan_description:
+                build_child_xml_node(vlan_root, 'description',
+                                                vlan_description)
+            intf_xml.append(vlan_root)
+        return intf_xml
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the xml necessary to remove the current configuration
+                  of the provided objects
+        """
+        intf_xml = []
+
+        if not want:
+            want = have
+
+        for config in want:
+            vlan_name = config['name']
+            vlan_root = build_root_xml_node('vlan')
+            vlan_root.attrib.update({'delete': 'delete'})
```

Comment:
@danielmellado Does `state_deleted` delete the vlan or remove configurations from the vlan when `config` is given in a task?

Reply:
it does delete the vlan, you can check what would happen in the deleted sample ;)